### PR TITLE
no pType2 on NDArrayConcat round2

### PR DESCRIPTION
**Pull request by akotlar**
bug, blocking release

### CODE REVIEW

File: hail/src/main/scala/is/hail/expr/ir/InferType.scala [hunk 1]
```diff
@@ -127,7 +127,7 @@ object InferType {
       case NDArrayReshape(nd, shape) =>
         TNDArray(coerce[TNDArray](nd.typ).elementType, Nat(shape.typ.asInstanceOf[TTuple].size), nd.typ.required)
       case NDArrayConcat(nds, _) =>
-        coerce[TStreamable](nds.typ).elementType
+        coerce[TStreamable](nds.typ).elementType.setRequired(true)
```
Comment (catoverdrive) [inline hail/src/main/scala/is/hail/expr/ir/InferType.scala]: Why did you change this to `setRequired(true)`? Did you mean to to create a TNDArray with a required element type?

Comment (akotlar) [reply hail/src/main/scala/is/hail/expr/ir/InferType.scala]: No, I saw this when I was debugging:

n.typ:     array<ndarray<int32, 2>>
n.ptype:   PCArray[PCNDArray[+PInt32,2]]
n.ptype2 +PCArray[+PCNDArray[+PInt32,2]]

x ptype:   PCNDArray[+PInt32,2]
x.ptype2 +PCNDArray[+PInt32,2]

(this is in Emit, beginning of NDArrayConcat)

This "fixed" the segfault, replaced with a type check error:

java.lang.UnsupportedOperationException
	at is.hail.expr.types.virtual.TNDArray._typeCheck(TNDArray.scala:97)
	at is.hail.expr.types.virtual.Type.typeCheck(Type.scala:242)
	at is.hail.TestUtils$.assertEvalsTo(TestUtils.scala:328)
	at is.hail.TestUtils$.assertEvalsTo(TestUtils.scala:305)
	at is.hail.expr.ir.IRSuite.assertNDEvals$1(IRSuite.scala:1944)

When I pushed I put the WIP label to avoid merging.

**Comment by catoverdrive** [comment]
@akotlar do you want to roll my changes into here and I can approve them, or should we just roll with #8187? I'm fine with either one, but we should probably close one of them.

**Comment by akotlar** [comment]
Let's go with your solution, thanks!
